include/trieste/driver.h

```diff
@@ -284,46 +292,45 @@ namespace trieste
 
           if (!prev || !wf)
           {
-            std::cout << "Skipping pass: " << pass->name() << std::endl;
+            logging::Info() << "Skipping pass: " << pass->name() << std::endl;
             continue;
           }
 
-          std::cout << "Testing pass: " << pass->name() << std::endl;
+          logging::Info() << "Testing pass: " << pass->name() << std::endl;
           wf::push_back(prev);
           wf::push_back(wf);
 
           for (size_t seed = test_seed; seed < test_seed + test_seed_count;
                seed++)
           {
-            std::stringstream ss1;
-            std::stringstream ss2;
-
             auto ast = prev.gen(parser.generators(), seed, test_max_depth);
-            ss1 << "============" << std::endl
+            logging::Trace() << "============" << std::endl
                 << "Pass: " << pass->name() << ", seed: " << seed << std::endl
                 << "------------" << std::endl
                 << ast << "------------" << std::endl;
 
-            if (test_verbose)
-              std::cout << ss1.str();
-
             auto [new_ast, count, changes] = pass->run(ast);
-            ss2 << new_ast << "------------" << std::endl << std::endl;
+            logging::Trace() << new_ast << "------------" << std::endl << std::endl;
 
-            if (test_verbose)
-              std::cout << ss2.str();
-
-            std::stringstream ss3;
-
-            auto ok = wf.build_st(new_ast, ss3);
-            ok = wf.check(new_ast, ss3) && ok;
+            auto ok = wf.build_st(new_ast);
+            ok = wf.check(new_ast) && ok;
 
             if (!ok)
             {
-              if (!test_verbose)
-                std::cout << ss1.str() << ss2.str();
-
-              std::cout << ss3.str() << "============" << std::endl
+              // Discussion Point
+              // COMMENT TO BE REMOVE BEFORE MERGE
```

Comment:
Isn't this an ideal opportunity for the "local variable" log option? I feel like I'm missing something.

Reply:
So we could do that, but then we would generate it on every test.  This is going to slow down fuzzing.  I am wondering if we should fail a fuzz, and then rerun it with a higher logging mode enabled, so that the run occurs again, with lots more information to aid in debugging.

Reply:
The initial AST has gone by the time we have realised it is failed, so that means stringing the AST on every pass.

Reply:
That seems reasonable